api/store/mongo/device_store.go

```diff
@@ -231,11 +231,17 @@ func (s *Store) DeviceRename(ctx context.Context, uid models.UID, name string) e
 func (s *Store) DeviceLookup(ctx context.Context, namespace, name string) (*models.Device, error) {
 	ns := new(models.Namespace)
 	if err := s.db.Collection("namespaces").FindOne(ctx, bson.M{"name": namespace}).Decode(&ns); err != nil {
+		if err == mongo.ErrNoDocuments {
+			return nil, store.ErrNoDocuments
+		}
```

Comment:
This is repeating all over the place. One thing we could do is to have a `fromMongoError(...)` method which could convert to the store error and avoid this duplication.